documentation/Authentication.md

```diff
@@ -0,0 +1,216 @@
+# Authentication
```

Comment:
This file is important. It shows different ways to authenticate. We should probably rename the file and it's content as node.js auth (since the repo will contain isomorphic js code).